From 17e93d7ce242e064f20ca4374ee6233a55bcf8a0 Mon Sep 17 00:00:00 2001
From: Thomas Betker <thomas.betker@rohde-schwarz.com>
Date: Mon, 11 May 2015 21:21:18 +0200
Subject: [PATCH 513/793] ARM: zynq: Turn on PL310 L2 cache prefetching

This patch was originally authored by John Linn on 22-Sep-2011 for
linux-xlnx.git, but got lost somehow. It turns on L2 cache prefetching
when XILINX_PREFETCH (formerly XILINX_L2_PREFETCH) is enabled.

The original patch also set the Auxiliary Control Register bit "Early
BRESP enable" (in addition to "Instruction prefetch enable" and "Data
prefetch enable"), but for Cortex-A9, this bit is now set automatically
by l2c310_enable().

Suggested-by: John Linn <john.linn@xilinx.com>
Signed-off-by: Thomas Betker <thomas.betker@rohde-schwarz.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 arch/arm/mach-zynq/common.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-zynq/common.c b/arch/arm/mach-zynq/common.c
index e305acd..06d6777 100644
--- a/arch/arm/mach-zynq/common.c
+++ b/arch/arm/mach-zynq/common.c
@@ -196,8 +196,13 @@ static const char * const zynq_dt_match[] = {
 
 DT_MACHINE_START(XILINX_EP107, "Xilinx Zynq Platform")
 	/* 64KB way size, 8-way associativity, parity disabled */
+#ifdef CONFIG_XILINX_PREFETCH
+	.l2c_aux_val    = 0x30400000,
+	.l2c_aux_mask   = 0xcfbfffff,
+#else
 	.l2c_aux_val    = 0x00400000,
 	.l2c_aux_mask	= 0xffbfffff,
+#endif
 	.smp		= smp_ops(zynq_smp_ops),
 	.map_io		= zynq_map_io,
 	.init_irq	= zynq_irq_init,
-- 
1.7.5.4

